core/src/main/java/org/elasticsearch/index/shard/DocsStats.java

```diff
@@ -49,7 +49,15 @@ public void add(DocsStats other) {
         if (other == null) {
             return;
         }
-        this.totalSizeInBytes += other.totalSizeInBytes;
+        if (this.totalSizeInBytes == -1) {
```

Comment:
if (this.totalSizeInBytes == -1) {
    this.totalSizeInBytes = other.totalSizeInBytes;
} else if (other.totalSizeInBytes != -1) {
    this.totalSizeInBytes += other.totalSizeInBytes;
}

Reply:
++ changed in https://github.com/elastic/elasticsearch/pull/27863/commits/68322921503038f8b837f17511bb0472f3856d8d